src/components/settings/Settings.cpp

```diff
@@ -23,25 +32,254 @@ void Settings::SaveSettings() {
 }
 
 void Settings::LoadSettingsFromFile() {
-  SettingsData bufferSettings;
+  MigrateSettingsToCBOR();
+
+  lfs_info cborFileInfo;
+  if (fs.Stat("/settings.cbor", &cborFileInfo) != LFS_ERR_OK) {
+    return;
+  }
+
+  uint8_t* cborBuf = new uint8_t[cborFileInfo.size]();
```

Comment:
instead of doing the memory management yourself use `std::vector`. You'll need to add `#include <vector>` at the includes and initialize the buffer with

```suggestion
  std::vector<uint8_t> cborBuf(cborFileInfo.size);
```

and when you need to pass a pointer to the fileRead function (or other functions) use `cborBuf.data()`

Then you won't have to do the `delete[] cborBuffer;` calls everywhere. The standard-library will do that for you

Reply:
Done, thank you.